os/mm/mm_heap/mm_check_heap_corruption.c

```diff
@@ -164,100 +174,107 @@ int mm_check_heap_corruption(struct mm_heap_s *heap)
 		 * Retake the semaphore for each region to reduce latencies
 		 */
 
-		if (!aborted
 #if defined(CONFIG_BUILD_FLAT) || defined(__KERNEL__)
-			&& !up_interrupt_context()
+		if (!abort_mode && !up_interrupt_context()) 
```

Comment:
If we are using abort_mode, then may be we should add CONFIG_BUILD_FLAT to the abort_mode definition as well